### PR TITLE
[Added] Hashicorp Vault authentication options

### PR DESCRIPTION
**Pull request by dangerousplay**
Hello, I hope you are having a good day.

Implemented a variety of Vault authentication mechanisms such as **Kubernetes**, **AppRole**, and **JWT**. These can be utilized when an explicit token is not provided, offering flexibility to the users which don't need to log in before on Vault.

Have a nice day :)

### CODE REVIEW

**Comment by PierreBeucher** [comment]
That's awesome, thanks ! I'll review it ASAP

**Comment by PierreBeucher** [comment]
It seems you're still actively working on this, I'll get on it once you pass PR in Ready state. Let me know if I can help :)

File: tests/test_hvault.rs [hunk 1]
```diff
@@ -186,7 +216,255 @@ async fn test_hashivault_client_address_load() -> Result<(), anyhow::Error> {
...
+
+    vault_configure_jwt_role(&client, jwt_mount, role, &token_claims).await?;
+
+    let token = jwt::encode(
+        &jwt::Header::new(jwt_alg),
+        &token_claims,
+        &jwt_key.to_encoding_key(),
+    )?;
+
+    env::set_var("VAULT_AUTH_JWT_TOKEN", token);
```
Comment (PierreBeucher) [inline tests/test_hvault.rs]: I can't make this test pass on my side, it seems this variable isn't used as `client.rs` will always look for file path locally. You probably wanted the client to use this variable somehow?

File: src/modules/hashivault/client.rs [hunk 2]
```diff
@@ -1,13 +1,20 @@
...
 use async_trait::async_trait;
 use std::{ collections::HashMap, env, fs, path::{Path, PathBuf} };
-use vaultrs::{kv2, kv1, aws, api::aws::requests::GenerateCredentialsRequest};
+use vaultrs::{kv2, kv1, aws, auth, api::aws::requests::GenerateCredentialsRequest};
 use log::debug;
 use home;
+use crate::modules::hashivault::config::{HashiVaultAuth};
+
+
+const KUBERNETES_SA_JWT_PATH: &str = "/var/run/secrets/kubernetes.io/serviceaccount/token";
```
Comment (PierreBeucher) [inline src/modules/hashivault/client.rs:15]: It seems this file is always used by default, ignoring the environment variable of the same name. Maybe you want to use env var here as well as default fle path?

File: src/modules/hashivault/config.rs [hunk 3]
```diff
@@ -13,6 +13,26 @@ pub struct HashiVaultInput {
 }
 
 
+#[derive(Debug, Deserialize, Clone, PartialEq, JsonSchema)]
+#[serde(tag = "type")]
+pub enum HashiVaultAuth {
```
Comment (PierreBeucher) [inline src/modules/hashivault/config.rs:18]: Is it possible to have an `Option<String>` here with the "usual" path? eg. `auth/kubernetes` for Kubernetes, etc. 

Not blocking for merge, I can update it myself later :)

**Comment by PierreBeucher** [comment]
LGTM for most things, thanks a lot for all these features ! A few things before merging:

- Can you rebase with [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) using `feat: xxx` and such? It will help generate release logs. I see you already did use similar convention, it should not be long :)
- Added a few comments for a failing test

**Comment by dangerousplay** [comment]
Thank you for the code review.

I will increment the code to support to informing the Vault auth role, AppRole role ID, mount path by environment variables. I will add default values for mount path.

File: src/modules/hashivault/client.rs (hunk 2 above)
Comment (dangerousplay) [reply src/modules/hashivault/client.rs:15]: Yes, I fixed the code to use this constant if the environment variable doesn't exists.

File: tests/test_hvault.rs (hunk 1 above)
Comment (dangerousplay) [reply tests/test_hvault.rs]: Yes, I fixed the code to use this constant if the environment variable doesn't exists.

File: src/modules/hashivault/config.rs (hunk 3 above)
Comment (dangerousplay) [reply src/modules/hashivault/config.rs:18]: Yes, I changed to use default values for the mount path.

**Comment by dangerousplay** [comment]
I hope you are having a good day ^^

I fixed the Vault authentication test that uses Kubernetes.
I changed the code to use default values for mount path depending on the authentication type. 
I rebased and squashed the commits using the conventional commit.

Sorry for the late update, unfortunately in the last week our city in Brazil was affected by one of the biggest water flood we have ever seen. Fortunately, we are secure and safe.

**Comment by PierreBeucher** [comment]
Hey, thanks for the fixes ! I'm sorry for this difficult situation you had, hopefully you're better now... Please don't be sorry, you made an awesome contribution and there is nothing to be late for.

All tests are now passing on my end. I have a few more checks to run before merging but LGTM :) Thanks again

**Comment by PierreBeucher** [comment]
Finally merged, sorry for the delay these last weeks I hadn't had much time for Novops. I'm gonna do a release soon with these new features, thanks again !

**Pull request merged**